lib/ansible/executor/task_executor.py

```diff
@@ -392,9 +399,10 @@ def _run_loop(self, items):
                         if var in task_vars and var not in self._job_vars:
                             del task_vars[var]
 
-        if task_fields is not None:
-            self._task.from_attrs(task_fields)
         self._task.no_log = no_log
+        # NOTE: run_once cannot contain loop vars because it's templated earlier also
+        # This is saving the post-validated field from the last loop so the strategy can use the templated value post task execution
+        self._task.run_once = task_fields.get('run_once')
```

Comment:
I was worried `self._task.from_attrs(task_fields)` might cause issues, so I tried to identify all the FieldAttributes potentially affected by this issue. I believe it's limited to the fields `run_once`, `ignore_unreachable`, `ignore_errors`, `loop`, and `no_log` since these may be used after the looped task is executed.

* `no_log` is fine, because it's set just above this change
* `loop` is fine because empty loops == skipped task and the codepath with the issue (template string is always truthy) doesn't occur
* `ignore_errors` is fine because the value for the Task is set to False if any of the individual items are False

So I fixed `ignore_unreachable` using the same pattern as `ignore_errors`, and limited this to just set `run_once`.